app/src/main/java/emu/skyline/settings/SettingsActivity.kt

```diff
@@ -102,6 +107,57 @@ class SettingsActivity : AppCompatActivity() {
             .commit()
     }
 
+    override fun onCreateOptionsMenu(menu : Menu?) : Boolean {
+        menuInflater.inflate(R.menu.settings_menu, menu)
+        val menuItem = menu!!.findItem(R.id.app_bar_search)
+        val searchView = menuItem.actionView as SearchView
+        searchView.queryHint = getString(R.string.search)
+
+        searchView.setOnQueryTextFocusChangeListener { _, focus ->
+            (binding.titlebar.toolbar.layoutParams as AppBarLayout.LayoutParams).scrollFlags =
+                if (focus)
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_NO_SCROLL
+                else
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL
+        }
+
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query : String) : Boolean {
+                return false
+            }
+
+            override fun onQueryTextChange(newText : String) : Boolean {
+                val queries = newText.split(",").toTypedArray()
+                if (newText.isNotEmpty()) {
+                    preferenceFragment.preferenceScreen.forEach { preferenceCategory ->
+                        val categoryVisible = queries.any { preferenceCategory.title?.contains(it, true) ?: false }
+                        // Tracks whether all preferences under this category are hidden
+                        var areAllPrefsHidden = true
+                        // Tracks whether there is at least one preference under this category that has a key
+                        var hasNonNullPreferenceKey = false
```

Comment:
The way I see it, "preferences" like Edge supporters and licenses should not appear as a search result as it would clutter the results, so the way I have to exclude them is this one, as they are entries that do not  have keys.

Reply:
Couldn't you filter them out ahead of time by checking if preferenceCategory has a key of category_credits or category_licenses?

Reply:
I could, but I would prefer not to hardcode stuff if possible. If settings ever change in the future again, that way the search will keep working.

Reply:
```
val documentationKeys = arrayOf("category_credits", "category_licenses")

override fun onCreateOptionsMenu(menu : Menu?) : Boolean {
...
preferenceFragment.preferenceScreen.forEach { preferenceCategory ->
    if (documentationKeys.contains(preferenceCategory.key)) {
        preferenceCategory.isVisible = false
        return@forEach
    }
...
```

It's hard-coded, sure, but also skips a lot of checking that each key exists and it's easy enough to maintain as keys change, get added, get deleted, etc.

Reply:
@AbandonedCart your example looks good, we're going with that